zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchSpanStore.java

```diff
@@ -63,6 +60,8 @@
     long beginMillis = Math.max(endMillis - request.lookback(), EARLIEST_MS);
 
     SearchRequest.Filters filters = new SearchRequest.Filters();
+    //only search for SERVER span,which is simpler and faster than aggregation
+    filters.addTerm("kind", "SERVER");
```

Comment:
we cannot assume this.. there are spans that are messaging in nature (PRODUCER/CONSUMER). also, RPC spans themselves are optional..

Reply:
i see,in out company,we use different queries ,for example in messaging circumstance,the kind may be messaging.UI splits them,not combine them all together

Reply:
aggregation is very slow in es especially in large indices